monai/transforms/post/array.py

```diff
@@ -248,3 +248,60 @@ def __call__(self, img):
             output = img
 
         return output
+
+
+class Contour(Transform):
+    """
+    Return the contour flag of organs in mask images that only compose of 0 and 1, with Laplace kernel
+        set as default for edge detection.
+
+    Args:
+        kernel_type: the method applied to do edge detection.
+    """
+
+    def __init__(self, kernel_type="Laplace"):
+        self.kernel_type = kernel_type
+
+    def __find_img_contour(self, img):
+        channels = img.shape[1]
+        conv = torch.nn.Conv2d(channels, channels, kernel_size=3, stride=1, padding=1, bias=False, groups=channels)
+        kernel = torch.tensor([[-1, -1, -1], [-1, 8, -1], [-1, -1, -1]], dtype=torch.float32, device=img.device)
+        kernel = kernel.repeat(channels, 1, 1, 1)
+        conv.weight = torch.nn.Parameter(kernel, requires_grad=False)
+
+        contour_img = conv(img)
+        torch.clamp_(contour_img, min=0.0, max=1.0)
+        return contour_img
+
+    def __call__(self, img):
+        """
+        Args:
+            img: torch tensor of the img that you want to find the contour of, with shape being
+                 (batch_size, channels, width, height[, depth])
+
+        Returns:
+            A torch tensor with the same dimention as img, note:
+                1. It's the binary classification result of whether a pixel is edge or not.
+                2. In order to keep the original shape of mask image, we use padding as default.
+                3. The edge detection is just approximate due to
+                    a) defects inherent to Laplace kernel, ideally the edge should be thin enough, but now it has a thickness.
+                    b) need to search the optimal/better thresold for classification
+        """
+        if self.kernel_type != "Laplace":
+            raise NotImplementedError
+        if img.ndim != 4 and img.ndim != 5:
+            raise RuntimeError("img.ndim should be 4 or 5")
+        if img.ndim == 4:
+            return self.__find_img_contour(img)
```

Comment:
why is the 2D case implemented as a separate private method?

Reply:
Will refactor in my PR together.
Thanks.

Reply:
because we are using different kernels for 2D and 3D